geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/MetadataFileSystemStorage.scala

```diff
@@ -97,24 +102,29 @@ abstract class MetadataFileSystemStorage(conf: Configuration,
   }
 
   override def getFilePaths(partition: String): java.util.List[Path] = {
-    import scala.collection.JavaConverters._
-
-    val partitionPath = StorageUtils.partitionPath(metadata.getRoot, partition)
-    val baseDir = if (metadata.getPartitionScheme.isLeafStorage) { partitionPath.getParent } else { partitionPath }
-
-    metadata.getFiles(partition).asScala.map(new Path(baseDir, _)).filter { p =>
-      if (PathCache.exists(metadata.getFileContext, p)) { true } else {
-        logger.warn(s"Inconsistent metadata for ${metadata.getSchema.getTypeName}: $p")
-        false
-      }
-    }.asJava
+    Option(metadata.getPartition(partition)).map(_.files()).filterNot(_.isEmpty) match {
+      case None => Collections.emptyList()
+      case Some(files) =>
+        val partitionPath = StorageUtils.partitionPath(metadata.getRoot, partition)
+        val baseDir = if (metadata.getPartitionScheme.isLeafStorage) { partitionPath.getParent } else { partitionPath }
+        val paths = new java.util.ArrayList[Path](files.size())
+
+        val iter = files.iterator
+        while (iter.hasNext) {
+          val path = new Path(baseDir, iter.next)
+          if (PathCache.exists(metadata.getFileContext, path)) {
+            paths.add(path)
+          } else {
+            logger.warn(s"Inconsistent metadata for ${metadata.getSchema.getTypeName}: $path")
```

Comment:
Is there something that the user/admin should do in this case?  (I.e., should one manage-metadata to rectify?)

Reply:
yeah, you could use `manage-metadata deregister` to remove the reference, or figure out why the file is missing and replace it